src/SIL.Harmony/Db/CrdtRepository.cs

```diff
@@ -9,34 +10,25 @@
 
 namespace SIL.Harmony.Db;
 
-internal class CrdtRepository(ICrdtDbContext _dbContext, IOptions<CrdtConfig> crdtConfig,
-    Commit? ignoreChangesAfter = null
-)
+internal class CrdtRepository
 {
-    private IQueryable<ObjectSnapshot> Snapshots => _dbContext.Snapshots.AsNoTracking();
+    private readonly ICrdtDbContext _dbContext;
+    private readonly IOptions<CrdtConfig> _crdtConfig;
 
-    private IQueryable<ObjectSnapshot> SnapshotsFiltered
+    public CrdtRepository(ICrdtDbContext dbContext, IOptions<CrdtConfig> crdtConfig,
+        Commit? ignoreChangesAfter = null)
     {
-        get
-        {
-            if (ignoreChangesAfter is not null)
-            {
-                return Snapshots.WhereBefore(ignoreChangesAfter, inclusive: true);
-            }
-            return Snapshots;
-        }
-    }
-    private IQueryable<Commit> Commits
-    {
-        get
-        {
-            if (ignoreChangesAfter is not null)
-            {
-                return _dbContext.Commits.WhereBefore(ignoreChangesAfter, inclusive: true);
-            }
-            return _dbContext.Commits;
-        }
+        _crdtConfig = crdtConfig;
+        _dbContext = ignoreChangesAfter is not null ? new ScopedDbContext(dbContext, ignoreChangesAfter) : dbContext;
+        //we can't use the scoped db context is it prevents access to the DbSet for the Snapshots,
+        //but since we're using a custom query, we can use it directly and apply the scoped filters manually
+        _currentSnapshotsQueryable = MakeCurrentSnapshotsQuery(dbContext, ignoreChangesAfter);
```

Comment:
I suppose you could instead add a `CurrentSnapshots` property to `ICrdtDbContext`, but I'm not sure what's better.
So, if you're happy 👍

Reply:
yeah I thought about that, but I'd rather keep all the queries in CrdtRepo, also `ICrdtDbContext` is provided by the application and they shouldn't be the ones to implement that query.